docs/proposals/dynamic-kubelet-settings.md

```diff
@@ -0,0 +1,129 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal for improving Kubelet config by versioning it and making it manageable via the API server.
+
+## Motivation
+
+The Kubelet is currently configured via command-line flags. This is painful for a number of reasons:
+- It makes it difficult to change the way Kubelets are configured in a running cluster, because it is often tedious to change the Kubelet startup configuration.
+- It makes it difficult to manage different Kubelet configurations for different nodes, e.g. if you want to canary a new config.
+- The current lack of versioned Kubelet configuration means that any time we change Kubelet flags, we risk breaking someone's setup.
```

Comment:
Here are some examples:

1. Heterogeneous machines which require different node allocatable or eviction settings
1. Nodes running different runtimes in the future.

Another use case if staged rollout of new kubelet features. For example disk eviction could be rolled out to a small percentage of machines before it hits the entire fleet. 

If the concern with per-node configs is config object duplication, we should probably address that.